packages/portal-proto/src/features/GenomicTables/SomaticMutationsTable/SMTableContainer.tsx

```diff
@@ -164,8 +167,46 @@ export const SMTableContainer: React.FC<SMTableContainerProps> = ({
         "gene.ssm.ssm_id",
       );
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [topSSM]);
+  }, [
+    data,
+    searchTerm,
+    previousSearchTerm,
+    topSSM,
+    previousTopSSM,
+    handleSurvivalPlotToggled,
+    searchTermsForGene?.geneSymbol,
+  ]);
+
+  useEffect(() => {
+    if (
+      topSSM &&
+      !isEqual(selectedSurvivalPlot, previousSelectedSurvivalPlot) &&
```

Comment:
Just a suggestion: If you use `useDeepCompareEffect` from `use-deep-compare`, you don't have to check these manually. We have been using this library for these scenarios.

Reply:
alright, using useDeepCompare now